### PR TITLE
MGDOBR-591: API mocks for SE instances

### PR DESCRIPTION
**Pull request by kelvah**
https://issues.redhat.com/browse/MGDOBR-591

Adding mocks for all the SE instance (bridge) endpoints.

To run the mocked version of the app:
- `npm run start:mocked`

<br>

<details>
  <summary>Fetch examples</summary>
  
<br>

Here some fetch examples to test the endpoints mocks, you can copy and paste them in the browser developer tools console. 
You'll have to replace `{bridgeId}` within the URL where present.

  
<br>

**Get all bridges**
```
fetch("https://event-bridge-event-bridge-prod.apps.openbridge-dev.fdvn.p1.openshiftapps.com/api/v1/bridges?page=0&size=10", {
  "method": "GET"  
});
```

  
<br>

**Get a bridge detail**
```
fetch("https://event-bridge-event-bridge-prod.apps.openbridge-dev.fdvn.p1.openshiftapps.com/api/v1/bridges/{bridgeId}", {
  "method": "GET"  
});
```
  
<br>

**Create a bridge**
```
fetch("https://event-bridge-event-bridge-prod.apps.openbridge-dev.fdvn.p1.openshiftapps.com/api/v1/bridges", {
  "headers": {
    "accept": "application/json",
    "content-type": "application/json"
  },
  "body": "{\"name\":\"new bridge\"}",
  "method": "POST",
  "mode": "cors"
});
```
  
<br>

**Delete a bridge**
```
fetch("https://event-bridge-event-bridge-prod.apps.openbridge-dev.fdvn.p1.openshiftapps.com/api/v1/bridges/{bridgeId}", {
  "method": "DELETE",
  "mode": "cors"
});
```
</details>

### CODE REVIEW

**Review by manstis — COMMENTED** [review]
You might want to implement the "de-provision" data flow too?

File: mocked-api/handlers.ts [hunk 1]
```diff
@@ -32,54 +34,180 @@ instancesData.map((instance) => {
...
+        id: {
+          equals: bridgeId as string,
+        },
+      },
+      data: {
+        status: "deprovision",
+      },
+    });
+
+    setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (kelvah) [reply mocked-api/handlers.ts:171]: > You might want to implement the "de-provision" data flow too?

Whoops. I'll look into it. Now that I'm thinking about it the deletion could also end up in the `failed` status. :sob:

**Comment by kelvah** [comment]
> You might want to implement the "de-provision" data flow too?

Is it OK if I do it within #30 ? So I can do it for both bridges and processors in one shot.

**Review by manstis — APPROVED** [review]

**Review by jomarko — COMMENTED** [review]
Is this PR dependent on the SSO #31 ? Askin because getting 401

Screenshot from 2022-04-26 10-43-31

**Comment by kelvah** [comment]
> Is this PR dependent on the SSO #31 ? Askin because getting 401
> 

No, it doesn't depend on #31. It should just work. 

I see from the screenshot that MSW reports there's no handler for the `bridges` endpoint. There should be. Let's have a chat to figure out what's going on.

**Comment by jomarko** [comment]
@kelvah , here I steps I did
- pulled your PR
- rebased with main
- npm install
- npm run start:mocked
- both in firefox and chrome on localhost:8080 I put commands from PR description into browser console

**Review by jomarko — APPROVED** [review]
Thank you, it works for me now!

**Pull request merged**